app/actions/remote/user.ts

```diff
@@ -287,6 +289,41 @@ export const fetchStatusInBatch = (serverUrl: string, id: string) => {
     return debouncedFetchStatusesByIds.apply(null, [serverUrl]);
 };
 
+const mentionNames = new Set<string>();
+export const fetchUserOrGroupsByMentionsInBatch = (serverUrl: string, mentionName: string) => {
+    mentionNames.add(mentionName);
+    return debouncedFetchUserOrGroupsByMentionNames.apply(null, [serverUrl]);
+};
+const debouncedFetchUserOrGroupsByMentionNames = debounce((serverUrl: string) => {
+    fetchUserOrGroupsByMentionNames(serverUrl, Array.from(mentionNames));
+}, 200, false, () => {
+    mentionNames.clear();
+});
+
+const fetchUserOrGroupsByMentionNames = async (serverUrl: string, mentions: string[]) => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const {database} = operator;
+
+    // Get any missing users
+    const usersInDb = await queryUsersByIdsOrUsernames(database, [], mentions).fetch();
+    const usersMap = new Set(usersInDb.map((u) => u.username));
+    const usernamesToFetch = mentions.filter((m) => !usersMap.has(m));
+    const {users} = await fetchUsersByUsernames(serverUrl, usernamesToFetch, false);
```

Comment:
Is it worth it to make here a size check before calling the fetch? (i.e. `if (usernamesToFetch.length) {...}`.